clientconn.go

```diff
@@ -467,15 +467,11 @@ func (cc *ClientConn) updateResolverState(s resolver.State) error {
 		return nil
 	}
 
-	if !cc.dopts.disableServiceConfig && cc.scRaw != s.ServiceConfig {
+	if !cc.dopts.disableServiceConfig {
 		// New service config; apply it.
-		sc, err := parseServiceConfig(s.ServiceConfig)
-		if err != nil {
-			fmt.Println("error parsing config: ", err)
-			return err
+		if sc, ok := s.ServiceConfig.(ServiceConfig); ok {
```

Comment:
Compare service config?
(Compare raw json? This doesn't work because of balancer configs, right?)